### DISCUSSION TITLE
No table and data extraction from pdf?

### DISCUSSION THREAD

**Discussion by Rapid1898-code** [jsvine/pdfplumber#788] [Ask for help with specific PDFs]
Hello - i would like to extract table data and text form the attached pdf - 

Normally i do this using eg. this code  below.
But with the attached pdf i simply get no table data and no text data.
Why is that?
(t first i thought the pdf has some images in it - but this seems the case with that pdf)

  ```
  pdf = pdfplumber.open(fn)    
    
    finalOutput= []
    workLines = []
    for idxPage, page in enumerate(pdf.pages, start=1): 
      if idxPage == 5:
        break                        
      # read the text from the tables and try to parse
      print(f"Reading tables for page {idxPage}")    
      erg=page.extract_tables(table_settings)
      for idxTable, table in enumerate(erg, start=1):
        for row in table:
          if row[0] in ["", None]:
            continue
          wRow = [idxPage, row[0]]
          worker = row[1:]
          worker = [x for x in worker if x not in ["",None]]
          worker = " ".join(worker)
          wRow.append(worker)
          finalOutput.append(wRow)        
    
      # read the text   line by line from the tables and try to parse
      workLines.extend(page.extract_text().splitlines())
      
    # for e in finalOutput:
    #   print(e)
    for l in workLines:
      print(l)
    exit()
```

[input.pdf](https://github.com/jsvine/pdfplumber/files/10429134/input.pdf)

**Comment by jsvine** [comment]
Hi @Rapid1898-code, and thanks for your interest in this library. Opening up the PDF in a standard PDF viewer, selecting all the text, and pasting it into a text editor produces something like this: 

<img width="222" alt="Screen Shot" src="https://user-images.githubusercontent.com/534702/212974073-6d85e4eb-fd4d-4b46-89f0-5f3e1d4f312b.png">

... so it appears that there's something off with the PDF's text/font-encodings. There may be ways to fix the PDF, but I don't have a specific recommendation for that, unfortunately.

**Comment by Rapid1898-code** [comment]
Ok thx - interesting...
